substrate/runtime/contract/src/vm.rs

```diff
@@ -0,0 +1,785 @@
+			GasMeterResult::Proceed => Ok(sandbox::ReturnValue::Unit),
+			GasMeterResult::OutOfGas => Err(sandbox::HostError),
+		}
+	}
+
+	// ext_put_storage(location_ptr: u32, value_non_null: u32, value_ptr: u32);
+	//
+	// Change the value at the given location in storage or remove it.
+	//
+	// - location_ptr: pointer into the linear
+	//   memory where the location of the requested value is placed.
+	// - value_non_null: if set to 0, then the entry
+	//   at the given location will be removed.
+	// - value_ptr: pointer into the linear memory
+	//   where the value to set is placed. If `value_non_null` is set to 0, then this parameter is ignored.
+	fn ext_set_storage<T: Trait, E: Ext<T>>(
+		e: &mut Runtime<T, E>,
+		args: &[sandbox::TypedValue],
+	) -> Result<sandbox::ReturnValue, sandbox::HostError> {
+		let location_ptr = args[0].as_i32().unwrap() as u32;
+		let value_non_null = args[1].as_i32().unwrap() as u32;
+		let value_ptr = args[2].as_i32().unwrap() as u32;
+
+		let mut location = [0; 32];
+
+		e.memory().get(location_ptr, &mut location)?;
+
+		let value = if value_non_null != 0 {
+			let mut value = [0; 32];
+			e.memory().get(value_ptr, &mut value)?;
+			Some(value.to_vec())
+		} else {
+			None
+		};
+		e.ext.set_storage(&location, value);
+
+		Ok(sandbox::ReturnValue::Unit)
+	}
+
+	// ext_get_storage(location_ptr: u32, dest_ptr: u32);
+	//
+	// Retrieve the value at the given location from the strorage.
+	// If there is no entry at the given location then all-zero-value
+	// will be returned.
+	//
+	// - location_ptr: pointer into the linear
+	//   memory where the location of the requested value is placed.
+	// - dest_ptr: pointer where contents of the specified storage location
+	//   should be placed.
+	fn ext_get_storage<T: Trait, E: Ext<T>>(
+		e: &mut Runtime<T, E>,
+		args: &[sandbox::TypedValue],
+	) -> Result<sandbox::ReturnValue, sandbox::HostError> {
+		let location_ptr = args[0].as_i32().unwrap() as u32;
+		let dest_ptr = args[1].as_i32().unwrap() as u32;
+
+		let mut location = [0; 32];
+		e.memory().get(location_ptr, &mut location)?;
+
+		if let Some(value) = e.ext.get_storage(&location) {
+			e.memory().set(dest_ptr, &value)?;
+		} else {
+			e.memory().set(dest_ptr, &[0u8; 32])?;
+		}
+
+		Ok(sandbox::ReturnValue::Unit)
+	}
+
+	// ext_transfer(transfer_to: u32, transfer_to_len: u32, value_ptr: u32, value_len: u32)
+	fn ext_transfer<T: Trait, E: Ext<T>>(
+		e: &mut Runtime<T, E>,
+		args: &[sandbox::TypedValue],
+	) -> Result<sandbox::ReturnValue, sandbox::HostError> {
+		let transfer_to_ptr = args[0].as_i32().unwrap() as u32;
+		let transfer_to_len = args[1].as_i32().unwrap() as u32;
+		let value_ptr = args[2].as_i32().unwrap() as u32;
+		let value_len = args[3].as_i32().unwrap() as u32;
+
+		let mut transfer_to = Vec::new();
+		transfer_to.resize(transfer_to_len as usize, 0);
+		e.memory().get(transfer_to_ptr, &mut transfer_to)?;
+		let transfer_to = T::AccountId::decode(&mut &transfer_to[..]).unwrap();
```

Comment:
use `?` instead of `unwrap`?

Reply:
let's move this discussion in [this thread](https://github.com/paritytech/polkadot/pull/345#discussion_r203060428).